staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -287,7 +306,9 @@ func PollImmediateInfinite(interval time.Duration, condition ConditionFunc) erro
 // PollUntil always waits interval before the first run of 'condition'.
 // 'condition' will always be invoked at least once.
 func PollUntil(interval time.Duration, condition ConditionFunc, stopCh <-chan struct{}) error {
-	return WaitFor(poller(interval, 0), condition, stopCh)
+	ctx, cancel := contextForChannel(stopCh)
+	defer cancel()
+	return WaitFor(poller(interval, 0), condition, ctx.Done())
```

Comment:
i'd suggest add a new func supporting passing context instead of stop channel as its signal since that long-termly we will support context in those wait utilities.

```go

func PollUntilWithContext(interval time.Duration, condition ConditionFunc, parentContext context.Context) {
        ctx, cancel := context.WithCancel(parentContext)
	defer cancel()
	return WaitFor(poller(interval, 0), condition, ctx.Done())
}


func PollUntil(interval time.Duration, condition ConditionFunc, stopCh <-chan struct{}) {
        ctx := stopChannelAsContext(stopCh)
        PollUntilWithContext(interval, condition, ctx)
} 

func stopChannelAsContext(parentCh <-chan struct{}) context.Context {
        ctx, cancel := context.WithCancel(context.Background())
 	go func() {
		select {
		case <-parentCh:
			cancel()
		case <-ctx.Done():
		}
	}()
	return ctx
}

```

Reply:
@yue9944882 thanks for the suggestion - it seems to me that work might be out of scope for this fix and should be addressed as part of a larger refactor to add context-based apis to _wait_ across the board, in addition to modifying known call sites. What do you think?